Sources/StreamChatUI/ChatMessageList/Attachments/Poll/PollCreationVC/PollCreationVC.swift

```diff
@@ -0,0 +1,685 @@
+        let isAnonymousPollSupported = pollsConfig.anonymousPoll.configurable
+
+        multipleVotesFeature.maxVotesConfig = isMaxVotesSupported ? MaximumVotesConfig(
+            isEnabled: pollsConfig.maxVotesPerPerson.defaultValue,
+            maxVotes: nil
+        ) : nil
+        multipleVotesFeature.isEnabled = isMultipleVotesSupported ? pollsConfig.multipleVotes.defaultValue : false
+        commentsFeature.isEnabled = isCommentsSupported ? pollsConfig.addComments.defaultValue : false
+        suggestionsFeature.isEnabled = isSuggestionsSupported ? pollsConfig.suggestAnOption.defaultValue : false
+        anonymousFeature.isEnabled = isAnonymousPollSupported ? pollsConfig.anonymousPoll.defaultValue : false
+    }
+
+    // MARK: - Update Content
+
+    override open func updateContent() {
+        super.updateContent()
+
+        validatePollOptions()
+        createPollButton.isEnabled = canCreatePoll
+    }
+
+    /// Validates if the poll options contain any errors.
+    open func validatePollOptions() {
+        optionsErrorIndices = [:]
+        options.enumerated().forEach { offset, option in
+            guard let optionsSectionIndex = sections.firstIndex(of: .options) else { return }
+            let indexPath = IndexPath(item: offset, section: optionsSectionIndex)
+            let cell = collectionView.cellForItem(at: indexPath) as? PollCreationOptionCell
+            var otherOptions = options
+            otherOptions.remove(at: offset)
+            if !option.isEmpty && otherOptions.contains(where: { $0 == option }) {
+                optionsErrorIndices[offset] = L10n.Polls.Creation.alreadyAnOptionError
+            }
+            cell?.content?.errorText = optionsErrorIndices[offset]
+        }
+    }
+
+    // MARK: - Presentation Delegate
+
+    open func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
+        !hasChanges
+    }
+
+    open func presentationControllerDidAttemptToDismiss(_ presentationController: UIPresentationController) {
+        showDismissConfirmation()
+    }
+
+    // MARK: - Actions
+
+    /// The cancel button was tapped.
+    @objc open func cancelPoll() {
+        if hasChanges {
+            showDismissConfirmation()
+            return
+        }
+        dismiss(animated: true)
+    }
+
+    /// Creates the poll with the current configuration.
+    @objc open func createPoll() {
+        createPollButton.isEnabled = false
+        channelController.createPoll(
+            name: name,
+            allowAnswers: commentsFeature.isEnabled,
+            allowUserSuggestedOptions: suggestionsFeature.isEnabled,
+            description: nil,
+            enforceUniqueVote: !multipleVotesFeature.isEnabled,
+            maxVotesAllowed: multipleVotesFeature.maxVotesConfig?.maxVotes,
+            votingVisibility: anonymousFeature.isEnabled ? .anonymous : .public,
+            options: options
+                .filter { !$0.isEmpty }
```

Comment:
Should we still trim the user input here?

Reply:
I rather not handle that for now. First, we don't know if customers want options to be trimmed; besides that, it makes finding duplicates more tricky. So for now, I think it is better we don't handle this

Reply:
Discussed internally. We created a ticket to tackle trimming options in the future.